### PR TITLE
VMD version 1.9.3

### PR DESCRIPTION
**Pull request by cass00**
To fix: location of patch file; elog message; metadata?
The patch file can be found under https://github.com/cass00/gentoo_vmd_patch

I have not checked  the minimal requirements for the dependencies, they might have to be increased.

### CODE REVIEW

File: sci-chemistry/vmd/vmd-1.9.3.ebuild [hunk 1]
```diff
@@ -56,6 +56,7 @@ QA_FLAGS_IGNORED_amd64=" usr/lib64/vmd/plugins/LINUX/tcl/volutil1.3/volutil"
 QA_FLAGS_IGNORED_x86=" usr/lib/vmd/plugins/LINUX/tcl/volutil1.3/volutil"
 
 pkg_nofetch() {
+  #This has to be FIXED!!!!
```
Comment (jlec) [inline sci-chemistry/vmd/vmd-1.9.3.ebuild]: What needs to be fixed here?

File: sci-chemistry/vmd/vmd-1.9.3.ebuild [hunk 2]
```diff
@@ -11,7 +11,7 @@ inherit cuda eutils flag-o-matic multilib prefix python-single-r1 toolchain-func
 DESCRIPTION="Visual Molecular Dynamics"
 HOMEPAGE="http://www.ks.uiuc.edu/Research/vmd/"
 SRC_URI="
-	https://dev.gentoo.org/~jlec/distfiles/${P}-gentoo-patches.tar.xz
+	${P}-gentoo-patches.tar.xz
```
Comment (jlec) [inline sci-chemistry/vmd/vmd-1.9.3.ebuild]: Could you please provide the updated patch set? I will host it again.

Comment (cass00) [inline sci-chemistry/vmd/vmd-1.9.3.ebuild]: The patch file can be found under https://github.com/cass00/gentoo_vmd_patch

Comment (jlec) [inline sci-chemistry/vmd/vmd-1.9.3.ebuild]: I pulled from your repo and hosted it on dev.gentoo.org.

File: sci-chemistry/vmd/vmd-1.9.3.ebuild (hunk 1 above)
Comment (cass00) [inline sci-chemistry/vmd/vmd-1.9.3.ebuild]: the location of the patch file (if changed)

**Comment by cass00** [comment]
What do you mean by not drop version 1.9.2?

**Comment by jlec** [comment]
> What do you mean by not drop version 1.9.2?

You did a `git mv`, didn't you? Just copy it.

**Comment by cass00** [comment]
I copied and then deleted ;). I had a temporary repo for the update (not the main gentoo one). If you can edit my pull request then just add the old ebuild file

**Comment by jlec** [comment]
@cass00 please add the old again, squash it into you current commit and force push it again. Okay?

**Comment by jlec** [comment]
When downloading the tarball for vmd it seems to be only tarred but not gzipped. @cass00 can you confirm this?

**Comment by cass00** [comment]
No idea. How do I check this?

**Comment by jlec** [comment]
I am missing USE support for avx512, libpng, liboptix, libospray, egl. Please diff the configure files and add those options.

Also the Manifest is missing the checksums for 1.9.2

**Comment by jlec** [comment]
Also cuda > 8 is needed.

File: sci-chemistry/vmd/vmd-1.9.3.ebuild [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+
+PYTHON_COMPAT=( python2_7 )
+
+inherit cuda eutils flag-o-matic multilib prefix python-single-r1 toolchain-funcs
+
+DESCRIPTION="Visual Molecular Dynamics"
+HOMEPAGE="http://www.ks.uiuc.edu/Research/vmd/"
+SRC_URI="
+	${P}-gentoo-patches.tar.xz
+	${P}.src.tar.gz"
```
Comment (junghans) [inline sci-chemistry/vmd/vmd-1.9.3.ebuild]: Btw, `http://www.ks.uiuc.edu/Research/vmd/${P}/files/final/${P}.src.tar.gz` should just work.

Comment (cass00) [inline sci-chemistry/vmd/vmd-1.9.3.ebuild]: Works indeed. Should this be used or does it create license problems?

**Comment by cass00** [comment]
avx512, libpng seem to work fine
liboptix needs https://developer.nvidia.com/optix
libospray needs this https://github.com/ospray/OSPRay/wiki
Even if you had these packages installed manually, I guess there would be some tweaking in the configure file necessary. Should they be included anyway?
egl seems to be mutually exclusive with openglpbuffer 
`OpenGLPbufferDisplayDevice.h:31:2: error: #error Cannot enable both EGL and GLX Pbuffer code at the same time (yet)`
However removing OPENGLPBUFFER from myconf breaks with 
`usr/lib/gcc/x86_64-pc-linux-gnu/4.9.4/../../../../x86_64-pc-linux-gnu/bin/ld: cannot find -lOpenGL`

**Comment by cass00** [comment]
Also, msms is not optional anymore according to configure file. Should the use flag be removed and sci-chemistry/msms-bin added as a "unconditional" dependency?

File: sci-chemistry/vmd/vmd-1.9.3.ebuild (hunk 3 above)
Comment (junghans) [inline sci-chemistry/vmd/vmd-1.9.3.ebuild]: @gentoo/licenses, I guess point 4. of the VMD license:
```
Licensee agrees to provide the Software only to any other person or entity who has registered with Illinois.
```

**Comment by jlec** [comment]
@cass00 yes regarding msms

File: sci-chemistry/vmd/vmd-1.9.3.ebuild (hunk 3 above)
Comment (jlec) [inline sci-chemistry/vmd/vmd-1.9.3.ebuild]: @junghans that's the reason why we need the fetch restriction.

**Comment by mgorny** [comment]
Ping. @cass00, there are unsolved issues reported by the CI and requests from reviewers. Are you still interested in doing this?

**Comment by cass00** [comment]
I have an ebuild without liboptix, libospray and egl that works fine. Haven't gotten any answers on my earlier questions.

**Comment by monsieurp** [comment]
@cass00 Please file a new PR if you're still interested.

**Pull request closed without merging**